src/common/action-button.js

```diff
@@ -47,19 +50,29 @@ export default class ActionButton extends Component {
         btnStyle,
         children,
         className,
+        disabled,
+        form,
         icon,
         size: bsSize,
-        style
+        style,
+        type = 'button'
       },
       state: { error, working }
     } = this
 
     return <Button
       bsStyle={error ? 'warning' : btnStyle}
-      disabled={working}
-      onClick={this._execute}
-
-      {...{ bsSize, className, style }}
+      disabled={working || disabled}
+      onClick={type !== 'submit' && this._execute}
+      ref={type === 'submit' && (button => {
```

Comment:
Move into `componentDidMount()` and remove the listener in `componentWillUnmount()`.